src/CoreWCF.Http/tests/Authorization/AuthorizationStartup.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Threading.Tasks;
+using CoreWCF.Channels;
+using CoreWCF.Configuration;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace CoreWCF.Http.Tests.Authorization;
+
+public class AuthorizationStartup
+{
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddAuthorization(options =>
+        {
+            options.AddPolicy(Policies.Write,
+                policy => policy.RequireClaim("scope", new[] { DefinedScopes.Write, DefinedScopes.Admin }));
+            options.AddPolicy(Policies.AdminOnly,
+                policy => policy.RequireClaim("scope", new[] { DefinedScopes.Admin }));
+        });
+        services.AddServiceModelServices();
+    }
+
+    public void Configure(IApplicationBuilder app)
+    {
+        app.UseServiceModel(builder =>
+        {
+            builder.AddService<SecuredService>();
+            builder.AddServiceEndpoint<SecuredService, ISecuredService>(new BasicHttpBinding
+            {
+                Security = new BasicHttpSecurity
+                {
+                    Mode = BasicHttpSecurityMode.None,
+                    Transport = new HttpTransportSecurity
+                    {
+                        ClientCredentialType = HttpClientCredentialType.InheritedFromHost
+                    }
+                }
+            }, "/BasicWcfService/basichttp.svc");
+            builder.AddServiceEndpoint<SecuredService, ISecuredService>(new BasicHttpBinding
+            {
+                Security = new BasicHttpSecurity
+                {
+                    Mode = BasicHttpSecurityMode.Transport,
+                    Transport = new HttpTransportSecurity
+                    {
+                        ClientCredentialType = HttpClientCredentialType.InheritedFromHost
+                    }
+                }
+            }, "/BasicWcfService/basichttp.svc");
+
+        });
+    }
+}
+
+[ServiceContract]
+public interface ISecuredService
+{
+    [OperationContract]
+    string Default(string text);
+
+    [OperationContract]
+    string AdminOnly(string text);
+
+    [OperationContract]
+    Task<string> Write(string text);
+
+    [OperationContract]
+    Task<string> Generated(string text);
+}
+
+public partial class SecuredService
+{
+    public Task<string> Generated(string text) => Task.FromResult(text);
+}
+
+[ServiceBehavior(IncludeExceptionDetailInFaults = true)]
+public partial class SecuredService : ISecuredService
+{
+    // No attribute => defaults to the builtin default policy which is RequireAuthenticatedUser
+    public string Default(string text) => text;
+
+    [Authorize(Policy = Policies.AdminOnly)]
```

Comment:
So AuthZ policies also allow specifying the authentication schemes as part of the policy, I don't see any code or tests for that, so I assume this is something not supported for WCF + Authorize?

Reply:
Indeed that's no currently supported. you can see [there](https://github.com/CoreWCF/CoreWCF/pull/850/files#diff-b51abb1fc1d805a3936e07b10b0bad6bb889566195f11573ebcc8103aa1a565e) that we are explicitly passing null to `AuthenticateAsync`.

Reply:
We will now trigger build warnings when AuthenticationSchemes or Roles property are set